lib/logstash/filters/base.rb

```diff
@@ -145,6 +146,24 @@ def filter(event)
     raise "#{self.class}#filter must be overidden"
   end # def filter
 
+
+  # in 1.5.0 multi_filter is meant to be used in the generated filter function in LogStash::Config::AST::Plugin only
+  # and is temporary until we refactor the filter method interface to accept events list and return events list,
+  # just list in multi_filter see https://github.com/elastic/logstash/issues/2872.
+  # refactoring the filter method will mean updating all plugins which we want to avoid doing for 1.5.0.
+  #
+  # @param events [Array<LogStash::Event] list of events to filter
+  # @return [Array<LogStash::Event] filtered events and any new events generated by the filter
+  public
+  def multi_filter(events)
```

Comment:
hmm, we're actually calling filter multiple times, hence multi_filter... no? 

this method is only used in the generated filter function and is ephemeral per #2872 because it will change the signature to handle processing multiple events, can we just keep it as-is knowing it will go away soon? :P